rust_src/src/symbols.rs

```diff
@@ -404,5 +404,42 @@ pub fn symbol_value(symbol: LispSymbolRef) -> LispObject {
     }
     val
 }
+/// Non-nil if VARIABLE has a local binding in buffer BUFFER.
+/// BUFFER defaults to the current buffer.
+#[lisp_fn(min = "1")]
+pub fn local_variable_p(variable: LispObject, buffer: LispBufferOrCurrent) -> bool {
+    let buf: LispBufferRef = buffer.into();
+    let mut symbol: LispSymbolRef = variable.into();
```

Comment:
You can specify the `mut LispSymbolRef` in the function declaration.

Reply:
i kept `LispObject` as we need it in `variable.eq(local_var)`